File: javascript/array/iterasiTransformasi.js
const arr = [1, 2, 3, 4];

// forEach
arr.forEach(x => {
  console.log(x);
});

// map
const doubled = arr.map(x => x * 2); // [2, 4, 6, 8]

// filter
const even = arr.filter(x => x % 2 === 0); // [2, 4]

// reduce
const sum = arr.reduce((acc, x) => acc + x, 0); // 10

// some
const hasEven = arr.some(x => x % 2 === 0); // true

// every
const allPositive = arr.every(x => x > 0); // true

// find
const found = arr.find(x => x > 2); // 3

// findIndex
const index = arr.findIndex(x => x > 2); // 2

File: rust/array/iterasiTransformasi.rs
let v = vec![1, 2, 3, 4];

// for_each
v.iter().for_each(|x| {
    println!("{}", x);
});

// map
let doubled: Vec<i32> = v.iter().map(|x| x * 2).collect();

// filter
let even: Vec<i32> = v.iter().filter(|x| *x % 2 == 0).cloned().collect();

// reduce (fold)
let sum: i32 = v.iter().fold(0, |acc, x| acc + x);

// any (some)
let has_even = v.iter().any(|x| x % 2 == 0);

// all (every)
let all_positive = v.iter().all(|x| *x > 0);

// find
let found = v.iter().find(|x| **x > 2);

// position (findIndex)
let index = v.iter().position(|x| x > 2);